src/pytest_alembic/executor.py

```diff
@@ -50,21 +49,28 @@ def run_command(self, command, *args, **kwargs):
         return self.stdout.readlines()
 
 
-@dataclass(frozen=True)
+@dataclass
 class ConnectionExecutor:
-    connection: Connection
+    metadatas: Dict[str, MetaData] = field(default_factory=dict)
 
```

Comment:
can no longer cache/store the connection because it can change call-to-call on the same revision.